### DISCUSSION TITLE
Trimming breaks page navigation

### DISCUSSION THREAD

**Discussion by PhantomGamers** [amwx/FluentAvalonia#391] [Q&A] [answered]
I'm not positive if the title is an accurate description of what's going on behind the scenes, but I have an Avalonia 11/FluentAvalonia app (<https://github.com/PhantomGamers/SFP>) that works fine when compiled either without trimming or with TrimMode set to partial, however when TrimMode is set to full (the default value), the navigation view shows but it never actually navigates to a page (and the nav buttons don't navigate either).

I have managed to make a slightly more barebones reproduction of the issue [on this project](https://github.com/PhantomGamers/FATrimTest)

Where the build published with `dotnet publish -o publish/ -c Release --self-contained -p:TrimMode=partial` works correctly:
![image](https://github.com/amwx/FluentAvalonia/assets/844685/33661296-2bee-4c81-aabb-809f2154f06c)
![image](https://github.com/amwx/FluentAvalonia/assets/844685/ee7eab38-dc68-4375-a1f5-66d602d55922)

But with the default TrimMode value set, publishing with `dotnet publish -o publish/ -c Release --self-contained ` I just get a blank page:
![image](https://github.com/amwx/FluentAvalonia/assets/844685/8a1aa1fc-8b07-4164-a32b-5d46a34393d4)

I can't reproduce this issue with the sample app (both the current version and v2), but I have no idea what I'm doing wrong.

I'll keep looking into this myself but I figured I would ask here in case someone has any idea as I've been trying to figure this out for a while now. Thanks!

**Comment by amwx** [comment, marked as answer]
I had a quick look at your code, and you're navigating via `Frame.Navigate(Type)`. This utilizes `Activator.CreateInstance` (reflection) to create the page views, which is likely the reason.

You can look into `INavigationPageFactory` which hands over the page creation to you (also allowing passing a ViewModel instance instead of just a Type). This is how the new sample app works, and the Frame page has some example code and instructions on this. I think that should help prevent this issue.

**Comment by PhantomGamers** [comment]
That solved it, thanks a lot!

I did it a little differently by just passing the page instance directly so I didn't have to change my code that much but it works great!  

<https://github.com/PhantomGamers/SFP/commit/d1540a8736df41ac36cf36d74e890b093b817e24>
